### PR TITLE
support allowDisabledDaysRangeSelect flag and hold coosed days on rangeSelect

### PR DESCRIPTION
**Pull request by jafar-rezaei**
With this PR , `DatePicker` and `Calendar` components , will able to get a flag called `allowDisabledDaysRangeSelect` [default is false] which help user to choose a range even with disabled day ...

this will add a `hidden` input beside of main input element which holds a JSON of days .

### CODE REVIEW

File: website/yarn.lock [hunk 1]
```diff
@@ -6239,8 +6239,8 @@ mitt@^1.1.2:
   resolved "https://registry.yarnpkg.com/mitt/-/mitt-1.1.3.tgz#528c506238a05dce11cd914a741ea2cc332da9b8"
 
 mixin-deep@^1.2.0:
-  version "1.3.1"
-  resolved "https://registry.yarnpkg.com/mixin-deep/-/mixin-deep-1.3.1.tgz#a49e7268dce1a0d9698e45326c5626df3543d0fe"
+  version "1.3.2"
```
Comment (thebrodmann) [inline website/yarn.lock]: This is not related to PR.

File: yarn.lock [hunk 2]
```diff
@@ -5562,8 +5562,8 @@ mississippi@^3.0.0:
     through2 "^2.0.0"
 
 mixin-deep@^1.2.0:
-  version "1.3.1"
-  resolved "https://registry.yarnpkg.com/mixin-deep/-/mixin-deep-1.3.1.tgz#a49e7268dce1a0d9698e45326c5626df3543d0fe"
+  version "1.3.2"
```
Comment (thebrodmann) [inline yarn.lock]: This is not related to PR too.

File: test/utils.test.js [hunk 3]
```diff
@@ -72,4 +73,34 @@ describe('Utility Functions', () => {
     expect(putZero('1')).toBe('01');
     expect(putZero('12')).toBe('12');
   });
+
+  test('should create a range of days between to date', () => {
+    const startDate = { year: 1398, month: 6, day: 29 };
+    const endDate = { year: 1398, month: 7, day: 3 };
+    expect(createDateRangeDays(startDate, endDate, [])).toEqual([
```
Comment (thebrodmann) [inline test/utils.test.js]: It sounds to me like multiple-day selection instead of a range! Multiple-day selection API has been proposed in #22.

File: src/DatePickerInput.js [hunk 4]
```diff
@@ -41,9 +42,20 @@ const DatePickerInput = React.forwardRef(
       if (formatInputText()) return formatInputText();
       return isDayRange ? getSelectedRangeValue() : getSelectedDayValue();
     };
+    const getSelectedRangeAvailableDays = () => {
+      if (!selectedDayRange.from || !selectedDayRange.to) return '[]';
```
Comment (thebrodmann) [inline src/DatePickerInput.js]: Are those single quotes mark around the empty array intentional?

Comment (jafar-rezaei) [reply src/DatePickerInput.js]: Yeah, actually I use `JSON.stringify` to maintain array of days as value of hidden input. Maybe it should be better to find another way

File: website/yarn.lock (hunk 1 above)
Comment (jafar-rezaei) [reply website/yarn.lock]: I forgot to clone from develop branch

File: test/utils.test.js (hunk 3 above)
Comment (jafar-rezaei) [reply test/utils.test.js]: some kind of , but using range date selection to get it

**Comment by jafar-rezaei** [comment]
closed in order to #35
